### PR TITLE
QE-21 sb 04 question

### PR DESCRIPTION
**Pull request by ramishenouda48**
1. Updated the question and question group drag and drop functionality to be dragged from the question title instead of an icon
2. Add Question meatball menu
3. Added Question duplication
4. Added question deletion

question meatball menu

### CODE REVIEW

File: src/components/SideBar/SideBarRow.js [hunk 1]
```diff
@@ -54,6 +58,10 @@ function SideBarRow({
           className={classNames(
             'sidebar-row-title-container d-flex align-items-center'
           )}
+          {...(!dragIcon && providedValue)}
```
Comment (d-Pixie) [inline src/components/SideBar/SideBarRow.js:61]: This is bad form. If `!dragIcon` is `false` you will end up with a destruct on `false`. I would suggest you change how you set the variable above: 
```js
const hasProvidedValue = Boolean(dragIcon && provided?.dragHandleProps)
const providedValue = hasProvidedValue ? { ...provided.dragHandleProps } : {}
```
so you can just do `{...providedValue}` here instead.

**Review by d-Pixie — CHANGES_REQUESTED** [review]

File: src/components/SideBar/SideBar.scss [hunk 2]
```diff
@@ -42,6 +42,10 @@
   .sidebar-row-title-container {
     cursor: pointer;
   }
+
+  &:hover .sidebar-meatball-menu {
+    opacity: 1 !important;
```
[the remark this answers is not in the document]
Comment (ramishenouda48) [reply src/components/SideBar/SideBar.scss]: Thank you! updated :)

File: src/components/SideBar/SideBarRow.js (hunk 1 above)
Comment (ramishenouda48) [reply src/components/SideBar/SideBarRow.js]: that's a good tip about the `false` thing, thank you 😃
I have removed the `dragIcon`, it was part of the old code... we are not really using it anymore

File: src/components/SideBar/SideBarRow.js [hunk 3]
```diff
@@ -54,6 +58,10 @@ function SideBarRow({
...
+          style={{
+            cursor: !dragIcon ? 'grab' : 'pointer',
```
[the remark this answers is not in the document]
Comment (ramishenouda48) [reply src/components/SideBar/SideBarRow.js]: we are not using the `dragIcon` anymore so I have removed it and set the style directly in the `scss` file.

File: src/components/SurveyStructure/RowQuestionsList.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    focusAndScrollToQuestion(reorderedQuestion)
+  }
+
+  const focusAndScrollToQuestion = (question) => {
+    updateFocusedQuestion(question)
+    ScrollToQuestion(question)
+  }
+
+  const reorderQuestions = (questionsList, startIndex, endIndex) => {
```
[the remark this answers is not in the document]
Comment (ramishenouda48) [reply src/components/SurveyStructure/RowQuestionsList.js:49]: name naming 👍  updated. to `moveQuestion`.

**Pull request merged**